Ks network analysis report 1 (#213)
tari-labs/tari-university: src/network-analysis/probabilistic-attack/building_blocks.md

- [Methodology](#methodology)
   - [Notation Used](#notation-used)
   - [Crude Monte Carlo Simulation](#crude-monte-carlo-simulation)
     - [Programming the Simulation](#programming-the-simulation)
     - [Use of Random Numbers](#use-of-random-numbers)
     - [Computing Statistics](#computing-statistics)
   - [Statistical Calculation](#statistical-calculation)
     - [Binomial Distribution Formula](#binomial-distribution-formula)
     - [Hypergeometric Distribution Formula](#hypergeometric-distribution-formula)
     - [Summation Formula](#summation-formula) 
- [References](#references)
- [Appendices](#appendices)
   - [Appendix A: Definitions of Terms](#appendix-a-definitions-of-terms)
- [Contributors](#contributors) 

## Introduction

The paper discusses building blocks to aid in answering of questions regarding the Tari Network. In order to perform 
analysis and draw accurate conclusions, a set of skills and programs need to be built and honed. Among these, is an 
understanding of statistics. This report focuses on types of distribution and the deriving of their formulae and touches 
on the simulations, specifically the Monte Carlo Simulation. 

## Types of Distribution 

> There are 6 high-level types of distributions. These include discrete, continuous , mixed discrete/continuous, joint, 

Reviewer: ```suggestion
There are a few high-level types of distributions, and these include discrete, continuous , mixed discrete/continuous, joint, and non-numeric.  
```

non-numeric and miscellaneous. 

> When considering solving the probability of an attacker controlling the majority of nodes in the network, the various 

Reviewer: ```suggestion
When determining the probability of an attacker controlling the majority of nodes in the network, the various 
```

types of probability distributions of the specific circumstances and variables of the 
problem need to be analyzed. There are two categories of probability distribution: finite and infinite support [[1]]. 

> 

Reviewer: ```suggestion
All probability distributions can be classified as discrete probability distributions or as continuous probability distributions, depending on whether they define probabilities associated with discrete variables or continuous variables” [1x]. 
```
[1x] https://stattrek.com/probability-distributions/discrete-continuous.aspx

### Continuous Distribution 

> A continuous random variable is a random variable with an infinite and uncountable set and range of possible values [[11]].

Reviewer: ```suggestion
If a variable can take on any value between two specified values, it is called a continuous variable, and its probability distribution is called a discrete probability distribution [1x].
```

Probabilities of continuous random variables (*X*) are defined as the area under the curve of its Probability Density 
Function<sup>[def][pdf~]</sup> (PDF). Therefore only ranges of values can have a nonzero probability. The probability that 

> a continuous random variable equals some value is always zero [[11]].

Reviewer: ```suggestion
a continuous random variable equals some value is always zero [[11]]. A typical plot of a continuous distribution is depicted below (although specifically for a normal distribution.)
```

<p align="center"><img src="assets/distribution_plot_normal_weight_shade_middle.png" width="700" /></p>

### Discrete Distribution 

> Likewise, to understand the discrete distribution, a discrete random variable requires definition. A discrete random 

Reviewer: ```suggestion
A discrete variable, as opposed to a continuous variable, can only take on a finite number of values.  If a random variable is a discrete variable, its probability distribution is called a discrete probability distribution [1x].   
```

variable is a random variable that has countable values, such as a list of non-negative integers. A discrete 
distribution thus describes the probability of occurrence of each value of a discrete random variable [[11]].

With a discrete probability distribution, each possible value of the discrete random variable can be associated with a 

> nonzero probability. Thus, the presentation of a discrete probability distribution is often in tabular form [[11]].

Reviewer: ```suggestion
nonzero probability. Thus, the presentation of a discrete probability distribution is often in tabular form [[11]], or a bar graph.
```

<p align="center"><img src="assets/distribution_plot_poisson_shade_right_tail.png" width="700" /></p>

> 

Reviewer: ```suggestion
Here's an overview of the various distributions and how they related; 
![image](https://user-images.githubusercontent.com/55133541/66566528-79406b80-eb65-11e9-9c48-dbc20fbc4937.png)

```

Examples of discrete distribution with **finite** support <sup>[def][discrete_distribution_finite~]</sup>  include the following: 

> 

Reviewer: We will leave that for later ...

$$

### Summation Formula

Refering to *Example $3$*, the **Tally** is the sum of all the combinations where the bad nodes are in a majority.  

As a recap to the question, *what is the probability of selecting a majority of bad nodes from a total of $100$ nodes 
if the committee size is $10$*? This problem considers solving for the probability where there is a **majority** of bad 
nodes in the committee. This entails calculating the probabilities from the BFT threshold to the committee size. Thus, 
there needs to be a summation of individual probabilities in order to calculate the probability for selecting the majority of bad nodes. 

$$
P_{tot} = \sum_{i=T}^{n} P(N,m,n,i)
$$

In order to understand this, the table below provides some visual insight. 

| &nbsp;&nbsp;Committee Size&nbsp;&nbsp; | &nbsp;&nbsp;BFT Threshold&nbsp;&nbsp; | &nbsp;&nbsp;No of Steps&nbsp;&nbsp; |
| :------------------------------------: | :-----------------------------------: | :---------------------------------: |
|                   10                   |                   7                   |                  3                  |

## References

> 

Reviewer: Insert these 

[1x] https://stattrek.com/probability-distributions/discrete-continuous.aspx 
[2x] https://medium.com/@srowen/common-probability-distributions-347e6b945ce4 

That’s **if you included** "the definitions of continuous and discrete variables" and "the picture of plots of distributions " as suggested above.
